### ISSUE TITLE
Conflicting directionality of constraint and variable value diff

### ISSUE THREAD

**Issue by pgkirsch** [convexengineering/gpkit#1510]
When you diff two models, the directionality (+/-) is opposite between the Constraint Differences and the Relative/Absolute Differences. I read `sol2.diff(sol1)` as what is the diff going `sol1 --> sol2`. See in the trivial example below that the Constraint Difference makes it look like the additional constraint has gone away (when it's in fact the inverse) whereas the solution value has increased 100% (which it has).

```python
In [1]: from gpkit import Variable, Model                                                                                                                      

In [2]: x = Variable("x")                                                                                                                                      

In [3]: m1 = Model(x, [x >= 1])                                                                                                                                

In [4]: sol1 = m1.solve()                                                                                                                                      
Solving took 0.151 seconds.

In [5]: m2 = Model(x, [x >= 1, x >= 2])                                                                                                                        

In [6]: sol2 = m2.solve()                                                                                                                                      
Solving took 0.177 seconds.

In [7]: print(sol2.diff(sol1))                                                                                                                                 
Solution Diff
=============
(positive means the argument is smaller)

Constraint Differences
**********************
--- removed in argument
+++ added in argument

@@ -6,4 +6,3 @@
 Constraints
 -----------
  x >= 1
- x >= 2

**********************

Relative Differences |above 1%|
-------------------------------
x : +100.0%

```

Basically comes down to the slight contradiction between `(positive means the argument is smaller)` and `+++ added in argument`.

**Comment by bqpd** [comment]
oh good call! making a pr

**Comment by bqpd** [comment]
example now runs:

```python
In [1]: from gpkit import Variable, Model                                                                                                                      

In [2]: x = Variable("x")                                                                                                                                      

In [3]: m1 = Model(x, [x >= 1])                                                                                                                                

In [4]: sol1 = m1.solve()     

In [5]: m2 = Model(x, [x >= 1, x >= 2])                                                                                                                        

In [6]: sol2 = m2.solve()

In [7]: print(sol2.diff(sol1))
```
```
Using solver 'mosek_cli'
 for 1 free variables
  in 2 posynomial inequalities.
Solving took 0.0381 seconds.
Using solver 'mosek_cli'
 for 1 free variables
  in 3 posynomial inequalities.
Solving took 0.019 seconds.
Solution Diff
=============
(positive means the argument is smaller)

Constraint Differences
**********************
--- only in argument
+++ not in argument

@@ -6,3 +6,4 @@
 Constraints
 -----------
  x >= 1
+ x >= 2

**********************

Relative Differences |above 1%|
-------------------------------
x : +100.0%

Sensitivity Differences |above 0.1|
-----------------------------------
The largest is +0.
```

**Comment by pgkirsch** [comment]
Thanks @bqpd!

One thought about the supporting text. The `positive means the argument is smaller` and `only/not in argument` are helpful, but, in two very minor ways, also slightly confusing, partly due to ordering and partly due to text. 

`positive means...` appears at the top, presumably so you don't have to repeat it for each category of difference (Relative, Absolute, etc.), but it also doesn't apply to the Constraint Differences, which gets it's own "overriding" support text. My proposal would be to replace these three lines with just one that applies to all sub-tables, i.e. instead of the `positive means...` line, something like `Diff is from subject --> object, where subject is argument`. I'm not sure if that's too abstract, but do you see what I'm trying to get at? Maybe something like `Argument is the baseline solution` is clearer? The result would be:


```
Solution Diff
=============
Argument is the baseline solution

Constraint Differences
**********************

 Constraints
 -----------
  x >= 1
+ x >= 2

**********************

Relative Differences |above 1%|
-------------------------------
x : +100.0%

Sensitivity Differences |above 0.1|
-----------------------------------
The largest is +0.
```

It could even be more explicit: `For sol2.diff(sol1), sol1 is the baseline`, but that's something that could maybe be added to the doctoring instead?

Anyway, obviously super minor and feel free to disregard.

**Comment by bqpd** [comment]
That's a great point, but would make removed constraints confusing. :( I'll merge what I have for now, but will continue thinking on a top-line message to rule them all..

**Comment by pgkirsch** [comment]
I'm not sure I follow. 

If original/argument/baseline model is:
```python
    x >= 1,
    x >= 2,
```
and new model is 
```python
    x >= 1,
```
Wouldn't it just be:
```
Solution Diff
=============
Argument is the baseline solution

Constraint Differences
**********************

 Constraints
 -----------
  x >= 1
- x >= 2

**********************

```

**Comment by bqpd** [comment]
d'oh I should have tried writing it out, that does work well. will implement!

**Comment by pgkirsch** [comment]
Thanks!
